pkg/machinepool/machinepool.go

```diff
@@ -29,10 +32,46 @@ func NewMachinePoolService() MachinePoolService {
 	return &machinePool{}
 }
 
+// ListMachinePools lists all machinepools (or, nodepools if hypershift) in a cluster
+func (m *machinePool) ListMachinePools(r *rosa.Runtime, clusterKey string, cluster *cmv1.Cluster) error {
+	// Load any existing machine pools for this cluster
+	r.Reporter.Debugf("Loading machine pools for cluster '%s'", clusterKey)
+	isHypershift := cluster.Hypershift().Enabled()
+	var err error
+	var machinePools []*cmv1.MachinePool
+	var nodePools []*cmv1.NodePool
+	machinePools, err = r.OCMClient.GetMachinePools(cluster.ID())
+	if isHypershift {
+		nodePools, err = r.OCMClient.GetNodePools(cluster.ID())
+	}
+	if err != nil {
+		return err
+	}
```

Comment:
if `isHypershift` is true, the error returned by GetMachinePools is ignored.
```suggestion
	machinePools, err = r.OCMClient.GetMachinePools(cluster.ID())
	if err != nil {
		return err
	}
	if isHypershift {
		nodePools, err = r.OCMClient.GetNodePools(cluster.ID())
	}
	if err != nil {
		return err
	}
```

Reply:
Will make this change - the intention was that the error does not matter if hypershift is true, as we don't care about machinepools, so the error should be ignored. But, it isn't good, as it's an extra API call, so will change it